apps/web/src/components/AssessmentStepTracker/AssessmentStepTracker.test.tsx

```diff
@@ -167,24 +78,49 @@ describe("AssessmentStepTracker", () => {
     ).toBeInTheDocument();
     expect(
       screen.getByRole("link", {
-        name: `4. ${bookmarkedCandidate.user.firstName} ${bookmarkedCandidate.user.lastName}`,
+        name: `4. ${firstByName.user.firstName} No last name provided`,
+      }),
+    ).toBeInTheDocument();
+    expect(
+      screen.getByRole("link", {
+        name: `5. ${lastByFirstName.user.firstName} ${lastByFirstName.user.lastName}`,
+      }),
+    ).toBeInTheDocument();
+    expect(
+      screen.getByRole("link", {
+        name: `6. ${bookmarkedCandidate.user.firstName} ${bookmarkedCandidate.user.lastName}`,
       }),
     ).toBeInTheDocument();
   });
 
+  it("should group results by candidate", () => {
+    renderAssessmentStepTracker();
+
+    // Has two results but should only see one link for it
+    expect(
+      screen.getAllByRole("link", {
+        name: `1. ${unassessedCandidate.user.firstName} ${unassessedCandidate.user.lastName}`,
```

Comment:
If the same user did appear multiple times, wouldn't the number be different?

Reply:
okay, updated to exclude the number and check for all instances to ensure only one appears in [update test for grouping](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9022/commits/10bd87a3ff2bd50980f2225a857c18c8cb9e485b)